gfx_scene.h

```diff
@@ -1886,39 +1716,170 @@ class GfxSceneInternal
+            case VK_FORMAT_BC7_SRGB_BLOCK: return DXGI_FORMAT_BC7_UNORM_SRGB;
+            case VK_FORMAT_UNDEFINED:
+            default: return DXGI_FORMAT_UNKNOWN;
+            }
+        };
+        image_ref->data.resize(0);
+        image_ref->data.reserve(ktxTexture_GetDataSizeUncompressed((ktxTexture*)ktx_texture));
+        image_ref->channel_count = ktxTexture2_GetNumComponents(ktx_texture);
+        uint32_t temp;
+        ktxTexture2_GetComponentInfo(ktx_texture, &temp, &image_ref->bytes_per_channel);
+        if(ktxTexture2_NeedsTranscoding(ktx_texture))
+        {
+            // Need to determine number of texture channels
+            ktx_transcode_fmt_e tf;
+            switch(image_ref->channel_count)
+            {
+            case 1: tf = KTX_TTF_BC4_R; break;
+            case 2: tf = KTX_TTF_BC5_RG; break;
+            default: tf = KTX_TTF_BC7_RGBA; break;
+            }
+            result = ktxTexture2_TranscodeBasis(ktx_texture, tf, KTX_TF_HIGH_QUALITY);
+            if(result != KTX_SUCCESS)
+                return GFX_SET_ERROR(kGfxResult_InvalidOperation, "Unable to transcode image `%s': %s", asset_file, ktxErrorString(result));
+            image_ref->data.reserve(ktxTexture_GetDataSize((ktxTexture*)ktx_texture));
+            result = ktxTexture_IterateLevelFaces((ktxTexture*)ktx_texture, &IterateKtxImage, &image_ref);
+        }
+        else
+        {
+            result = ktxTexture_IterateLoadLevelFaces((ktxTexture*)ktx_texture, &IterateKtxImage, &image_ref);
+        }
+        if(result != KTX_SUCCESS)
+            return GFX_SET_ERROR(kGfxResult_InvalidOperation, "Unable to load image `%s': %s", asset_file, ktxErrorString(result));
+        image_ref->format = vk_to_dxgi(ktx_texture->vkFormat);
+        image_ref->width = ktx_texture->baseWidth;
+        image_ref->height = ktx_texture->baseHeight;
+        if(image_ref->format == DXGI_FORMAT_UNKNOWN)
+            image_ref->format = GetImageFormat(*image_ref);
+        image_ref->mips = ktx_texture->numLevels > 1;
+        image_ref->flags = (image_ref->channel_count != 4 ? 0 : kGfxImageFlag_AlphaChannel);
+
+        GfxMetadata& image_metadata = image_metadata_[image_ref];
+        image_metadata.asset_file = asset_file; // set up metadata
+        image_metadata.object_name = file;
+        ktxTexture_Destroy((ktxTexture*)ktx_texture);
+        return kGfxResult_NoError;
+    }
+
     GfxResult importImage(GfxScene const &scene, char const *asset_file)
     {
         GFX_ASSERT(asset_file != nullptr);
         int32_t image_width, image_height, channel_count;
         if(gfxSceneFindObjectByAssetFile<GfxImage>(scene, asset_file))
             return kGfxResult_NoError;  // image was already imported
-        stbi_uc *image_data = stbi_load(asset_file, &image_width, &image_height, &channel_count, 0);
+        stbi_uc *image_data = nullptr;
+        uint32_t bytes_per_channel = 2;
+        if(stbi_is_16_bit(asset_file))
```

Comment:
Also added support for loading 16bit images. Previously the tinygltf loader had 16b image support but this makes the functionality available to all.